### PR TITLE
Pd plugins

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 1]
```diff
@@ -1,30 +1,45 @@
-{ stdenv, fetchurl, unzip, puredata }:
+{ stdenv, fetchurl, unzip, puredata, fftwSinglePrec }:
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: Just `fftw`.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 2]
```diff
@@ -1,30 +1,45 @@
...
   name = "timbreid-${version}";
 
   src = fetchurl {
     url = "http://williambrent.conflations.com/pd/timbreID-${version}-src.zip";
-    sha256 = "02rnkb0vpjxrr60c3hryv7zhyjpci2mi9dk27kjxpj5zp26gjk0p";
+    sha256 = "14k2xk5zrzrw1zprdbwx45hrlc7ck8vq4drpd3l455i5r8yk4y6b";
   };
 
-  buildInputs = [ unzip puredata ];
+  buildInputs = [ unzip puredata fftwSinglePrec ];
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: `fftw`.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 3]
```diff
@@ -1,30 +1,45 @@
...
+
+  makeFlags = [
+    "CFLAGS= -I${puredata}/include/pd -Wall -W -g “-fPIC” --enable-float"
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: Do not need unnecessary flags. Add necessary flags with `NIX_CFLAGS_COMPILE`. It looks like `--enable-float` belongs in the configure flags, not here. What are those fancy quotes around `-fPIC` doing?

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 4]
```diff
@@ -1,30 +1,45 @@
...
+  ];
 
   unpackPhase = ''
+    mkdir source
+    cd source
     unzip $src
-    mv timbreID-0.6.0-src/tID/* .
-    rm -rf timbreID-0.6.0-src/tID/
-    rm -rf timbreID-0.6.0-src/INSTALL.txt
   '';
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: Just delete the unpackPhase.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 5]
```diff
@@ -1,30 +1,45 @@
...
 
+  buildPhase = ''
+    make tIDLib.o all
+ '';
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: Are those arguments needed? `tIDLib.o` looks useless. Specify needed arguments in `makeFlags`. By replacing the `buildPhase` you effectively ignore the value of `makeFlags`.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix [hunk 6]
```diff
@@ -1,30 +1,45 @@
...
+    cp -r audio/ $out/
+    cp -r data/ $out/
+    cp -r doc/ $out/
   '';
 
+  postFixup = ''
+    mv $out/share/doc/ $out/
+    rm -rf $out/share/
+    '';
+
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/timbreid/default.nix]: This is not needed. The default fixup phase moves `doc` in the right place`.

File: pkgs/applications/audio/pd-plugins/maxlib/default.nix [hunk 7]
```diff
@@ -1,28 +1,26 @@
...
 
-  src = fetchurl {
-    url = "mirror://sourceforge/project/pure-data/libraries/maxlib/${name}.tar.gz";
-    sha256 = "0vxl9s815dnay5r0067rxsfh8f6jbk61f0nxrydzjydfycza7p1w";
+  src = fetchFromGitHub {
+    owner = "electrickery";
+    repo = "pd-maxlib";
+    rev = "v${version}";
+    sha256 = "10w9qfgn26lj3zqjksf2r1wsjpf5xy4dx22jay9l6idy9q62mxsn";
   };
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/maxlib/default.nix]: This does not look like the official continuation of maxlib. For instance, `meta.homepage` below does not link here, and the developer is different. We may replace the source, but could you explain why?

File: pkgs/applications/audio/pd-plugins/cyclone/default.nix [hunk 8]
```diff
@@ -1,32 +1,44 @@
...
+    # for file in `grep -r -l m_imp.h`
+      # do
+        # sed -i 's|#include "m_imp.h"|#include "${puredata}/include/pd/m_imp.h"|g' $file
+      # done
+  # '';
+
+  # installPhase = ''
+    # mkdir -p $out/cyclone
+    # cp -r bin/* $out/cyclone
+  # '';
```
Comment (orivej) [inline pkgs/applications/audio/pd-plugins/cyclone/default.nix]: Just delete the commented out code.

**Review by orivej — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/pd-plugins/cyclone/default.nix (hunk 8 above)
Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/cyclone/default.nix]: OK, done. I'll push once everything else is OK too.

File: pkgs/applications/audio/pd-plugins/maxlib/default.nix (hunk 7 above)
Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/maxlib/default.nix]: Maybe I'm too gullable, but the readme.md of this source says:
``Please note that Olaf does not maintain this code any longer, so questions and bug-reports should be directed towards me.``
Furthermore it seems like he's only doing bugfixes.

Comment (orivej) [reply pkgs/applications/audio/pd-plugins/maxlib/default.nix]: OK.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix (hunk 4 above)
Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/timbreid/default.nix]: without the unpackphase I get:
```
unpacking source archive /nix/store/lw38kd4a9j3bdhamr4ngr8apz46bww5z-timbreID-0.7.0-src.zip
unpacker produced multiple directories
builder for ‘/nix/store/gvgkzyjpr7dmzc2naa58nr7ivh3b7l3l-timbreid-0.7.0.drv’ failed with exit code 1
```

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix (hunk 5 above)
Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/timbreid/default.nix]: without this, I get:
```
building
build flags: SHELL=/nix/store/ski5znw2b8bj5qnsncrcq6128hkrwgvq-bash-4.4-p12/bin/bash LIBDIR=\$\(out\)/lib
gcc -DPD -I/Applications/Pd-0.47-1.app/Contents/Resources/src -I/usr/local/include -Wall -W -g -fPIC -O6 -funroll-loops -fomit-frame-pointer -o "bark.o" -c "bark.c"
gcc  -Wl,--export-dynamic -shared -fPIC -o "bark.pd_linux" "bark.o" tIDLib.o -lfftw3f -lc
gcc: error: tIDLib.o: No such file or directory
make: *** [Makefile:57: bark.pd_linux] Error 1
rm bark.o
builder for ‘/nix/store/22cc8jl0ipn5y4sq3z89p8rbnz7mb6hg-timbreid-0.7.0.drv’ failed with exit code 2
```

Also, the install instructions tell you to do this.

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix (hunk 6 above)
Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/timbreid/default.nix]: it's actually the fixup phase that moves it from the right place to the wrong one, and this moves it back

**Comment by magnetophon** [comment]
@orivej better?  :)

File: pkgs/applications/audio/pd-plugins/timbreid/default.nix (hunk 6 above)
Comment (orivej) [reply pkgs/applications/audio/pd-plugins/timbreid/default.nix]: Then fixup is right.

Comment (magnetophon) [reply pkgs/applications/audio/pd-plugins/timbreid/default.nix]: @orivej I realize that normally docs go in $out/share/doc, but in this case some of the help patches point to stuff in $out/doc, so I moved it.

**Comment by magnetophon** [comment]
@orivej afaik I fixed all your issues.  What do you think?

**Comment by magnetophon** [comment]
@orivej Is there anything to be done left?

**Comment by orivej** [comment]
I would have investigated the changes requested in the comments (after your replies), but this PR is good enough to be merged regardless. Thank you!

**Pull request merged**